### PR TITLE
cli: Fix failure with "set-name" and "bridges"

### PR DESCRIPTION
**Pull request by mkj**
## Description
The code path when a config has `set-name` as well as `bridges` doesn't seem to work at present, failure below. This change seems to fix it. 

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [ ] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

Config:
```yaml
network:
  version: 2
#  renderer: networkd
#  networkd doesn't handle ethernet well...
  renderer: NetworkManager
  ethernets:
    devnet:
      renderer: networkd
      match:
        name: enx00aa11223344
    enx00bb11223344:
      addresses: [ 10.20.0.1/24 ]
      renderer: networkd
    wlan:
      match:
        macaddress: 00:99:11:22:33:44
      set-name: "wlan0"

  bridges:
    # ... and NetworkManager doesn't handle bridges well
    renderer: networkd
    br0:
      addresses: [ "10.9.0.1/16" ]
      interfaces: [ "devnet" ]
```

```
netplan try
Traceback (most recent call last):
  File "/usr/share/netplan/netplan/cli/commands/try_command.py", line 84, in command_try
    NetplanApply().command_apply(run_generate=True, sync=True, exit_on_error=False)
  File "/usr/share/netplan/netplan/cli/commands/apply.py", line 191, in command_apply
    changes = NetplanApply.process_link_changes(devices, config_manager)
  File "/usr/share/netplan/netplan/cli/commands/apply.py", line 281, in process_link_changes
    if NetplanApply.is_composite_member(composite_interfaces, phy):
  File "/usr/share/netplan/netplan/cli/commands/apply.py", line 252, in is_composite_member
    members = settings.get('interfaces', [])
AttributeError: 'str' object has no attribute 'get'
```

### CODE REVIEW

File: netplan/cli/commands/apply.py [hunk 1]
```diff
@@ -251,17 +251,16 @@ def command_apply(self, run_generate=True, sync=False, exit_on_error=True):  # p
...
     @staticmethod
-    def is_composite_member(composites, phy):  # pragma: nocover (covered in autopkgtest)
+    def is_composite_member(composites, phy):
         """
         Is this physical interface a member of a 'composite' virtual
         interface? (bond, bridge)
         """
         for composite in composites:
-            for _, settings in composite.items():
-                members = settings.get('interfaces', [])
```
Comment (slyon) [inline netplan/cli/commands/apply.py:261]: I think we should rather check something like this before `members = settings.get('interfaces', [])`:
```
if not type(settings) is dict:
    continue
```

**Review by slyon — CHANGES_REQUESTED** [review]
Thank you very much @mkj for your contribution!
This is/was an untested code-path, which is bad (from Netplan's side!). I've added a new testing class and corresponding unit tests for this method, to make sure it does not fail in the future. And pushed this unittest to your branch!

The unit test also shows that your implementation does not seem to work correctly, it always returns `False`, I've left an inline comment of what should be an easy fix for this issue.

What do you think?

File: netplan/cli/commands/apply.py (hunk 1 above)
Comment (mkj) [reply netplan/cli/commands/apply.py:261]: Yes that is a much more sensible fix, I had mistaken what the arguments were. Thanks

Comment (slyon) [reply netplan/cli/commands/apply.py:261]: Could you update that accordingly? As I do not want to do a self-review & merge of this change.

Comment (mkj) [reply netplan/cli/commands/apply.py:261]: I've pushed that change now.

**Review by slyon — APPROVED** [review]
Thank you for fixing up the comments I had.
This is ready for merging!

**Pull request merged**